### PR TITLE
feat: add emerging markets rule

### PR DESCRIPTION
**Pull request by vitalymatyushik**
Add "Emerging Markets" rule

[Related issue ](https://github.com/ghostfolio/ghostfolio/issues/3930)

image

### CODE REVIEW

File: apps/api/src/app/portfolio/portfolio.service.ts [hunk 1]
```diff
@@ -1180,6 +1181,11 @@ export class PortfolioService {
                   new AccountClusterRiskSingleAccount(
                     this.exchangeRateDataService,
                     accounts
+                  ),
+                  new AllocationClusterRiskEmergingMarkets(
```
Comment (dtslvr) [inline apps/api/src/app/portfolio/portfolio.service.ts:1185]: Move to a new section:
`allocationClusterRisk: summary.ordersCount > 0 ? await this.rulesService.evaluate([ ... ])`

File: apps/api/src/app/portfolio/portfolio.service.ts [hunk 2]
```diff
@@ -1160,7 +1161,7 @@ export class PortfolioService {
     const userId = await this.getUserId(impersonationId, this.request.user.id);
     const userSettings = <UserSettings>this.request.user.Settings.settings;
 
-    const { accounts, holdings, summary } = await this.getDetails({
+    const { accounts, holdings, summary, markets } = await this.getDetails({
```
Comment (dtslvr) [inline apps/api/src/app/portfolio/portfolio.service.ts:1164]: Reorder alphabetically please:
`const { accounts, holdings, markets, summary } = ...`

File: apps/api/src/app/user/user.service.ts [hunk 3]
```diff
@@ -215,6 +216,10 @@ export class UserService {
         undefined,
         {}
       ).getSettings(user.Settings.settings),
+      AllocationClusterRiskEmergingMarkets:
+        new AllocationClusterRiskEmergingMarkets(undefined, 0, 0).getSettings(
```
Comment (dtslvr) [inline apps/api/src/app/user/user.service.ts:220]: `undefined, undefined, undefined`

File: apps/api/src/models/rules/allocation-cluster-risk/emerging-markets.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      ? this.emergingMarketsValueInBaseCurrency / this.totalInvestment
+      : 0;
+
+    if (emergingMarketsValueRatio > ruleSettings.thresholdMax) {
+      return {
+        evaluation: `The emerging markets contribution exceed ${(
+          ruleSettings.thresholdMax * 100
+        ).toPrecision(
+          3
+        )}% of your initial investment (${(emergingMarketsValueRatio * 100).toPrecision(3)}%)`,
```
Comment (dtslvr) [inline apps/api/src/models/rules/allocation-cluster-risk/emerging-markets.ts:36]: of your **current** investment

File: apps/api/src/models/rules/allocation-cluster-risk/emerging-markets.ts [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        value: false
+      };
+    } else if (emergingMarketsValueRatio < ruleSettings.thresholdMin) {
+      return {
+        evaluation: `The emerging markets contribution does not exceed ${(
+          ruleSettings.thresholdMin * 100
+        ).toPrecision(
+          3
+        )}% of your initial investment (${(emergingMarketsValueRatio * 100).toPrecision(3)}%)`,
```
Comment (dtslvr) [inline apps/api/src/models/rules/allocation-cluster-risk/emerging-markets.ts:45]: of your **current** investment

**Review by dtslvr — CHANGES_REQUESTED** [review]
Thanks for this contribution, @vitalymatyushik! 😎

As this is the first rule with `thresholdMin` and `thresholdMax`, it is a bit more tricky. I have added a few comments and hope that clarifies it.

**Comment by vitalymatyushik** [comment]
thank you @dtslvr!
I've addressed the comments and added a new section 

image

**Review by dtslvr — APPROVED** [review]
This is a great addition! 🙌🏻 Thank you, @vitalymatyushik!
Ready to merge.

**Pull request merged**